code/game/objects/items/devices/scanners/gas_analyzer.dm

```diff
@@ -169,7 +171,7 @@
 		else
 			message += airs.len > 1 ? span_notice("This node is empty!") : span_notice("[target] is empty!")
 		
-		if(tool)
+		if(valid_analyzer)
 			tool.last_gasmix_data += list(gas_mixture_parser(air, mix_name))
```

Comment:
This smells a bit off, though I do understand why you're doing it the way you are, I would very much prefer just letting the caller handle setting this list checking like this feels really nasty.

At the very least, why not build a list of gasmix data, and then do the istype check. The cost is minimal, and you don't need to futz around quite so much

Reply:
Can move it to
https://github.com/tgstation/tgstation/blob/58a2a979d3dcab2e7646c0721c40eb0d3b423de3/code/game/objects/objs.dm#L359
and 
https://github.com/tgstation/tgstation/blob/58a2a979d3dcab2e7646c0721c40eb0d3b423de3/code/game/objects/items/devices/scanners/gas_analyzer.dm#L116

Want me to do it?

Reply:
Actually on second thought that's also difficult. Ill do the second one.